script.js

```diff
@@ -1,88 +1,69 @@
 let books = [
-    {
-   titulo: '100 años de soledad',
-   author: 'Gabriel Garcia Marquez'
-    },
-    {
-   titulo: 'El tirano',
-   author: 'Massimo Manfredy'
-    }
-]
-
-const bookList = document.querySelector('.books-display');
-let displ = ''
-books.forEach((el, index) => {    
-    displ += `
-    <div class="oneBook" id="book${index}">
-    <h4> ${el.titulo} </h4>
-    <h4> ${el.author} </h4> 
-    <button class="btn-display" id="${index}">remove</button> 
-    </div>
-    ` 
-
-});
-bookList.innerHTML = displ
-let removIds = [...document.querySelectorAll('.btn-display')].map(item => item.id)
-
-const newB = document.getElementById('bk')
-
-newB.addEventListener('submit', function(event){
+  {
+    titulo: '100 años de soledad',
+    author: 'Gabriel Garcia Marquez',
+  },
+  {
+    titulo: 'El tirano',
+    author: 'Massimo Manfredi',
+  },
+];
+
+function storage(ok) {
+  window.localStorage.setItem('localBooks', JSON.stringify(ok));
+}
 
-    event.preventDefault();
-    const reciTi = document.getElementById("recibe-ti").value
-    const reciAu = document.getElementById("recibe-au").value
+function getBooks() {
+  books = JSON.parse(window.localStorage.getItem('localBooks'));
+}
 
-    if(reciTi !== '' && reciAu !== ''){
-    let newBook = {
-        titulo: reciTi,
-        author: reciAu
-    }
-    books.unshift(newBook)
+getBooks();
 
-    let displ = ''
-    books.forEach((el, index) => {    
+function display() {
+  const bookList = document.querySelector('.books-display');
+  let displ = '';
+  books.forEach((el, index) => {
     displ += `
     <div class="oneBook" id="book${index}">
     <h4> ${el.titulo} </h4>
     <h4> ${el.author} </h4> 
     <button class="btn-display" id="${index}">remove</button> 
     </div>
-    `
-
-    });
-    bookList.innerHTML = displ
-
-    console.log(books)
+    `;
+  });
+  bookList.innerHTML = displ;
 }
-})
-
-/* document.querySelectorAll('.btn-display').forEach( (rem) => {
-    rem.addEventListener('click', (event) => {
-        console.log(rem)
-    })
 
-}) */
+// ---------ADD-------------
+
+const newB = document.getElementById('bk');
+newB.addEventListener('submit', (event) => {
+  event.preventDefault();
+  const reciTi = document.getElementById('recibe-ti').value;
+  const reciAu = document.getElementById('recibe-au').value;
+
+  if (reciTi !== '' && reciAu !== '') {
+    const newBook = {
+      titulo: reciTi,
+      author: reciAu,
+    };
+    books.unshift(newBook);
+    storage(books);
+    display();
+  }
```

Comment:
- [ ] Great job so far 👏. I would like to inform you that despite all improvements you have made, your functions to add and remove book items are not working.  I tried to run it on my local machine in the browser and this is the error I got. please try to correct it so that It can work 👍 
![image](https://user-images.githubusercontent.com/90222110/182436098-7086fc0d-a93f-49c5-b858-2d1a33145580.png)

Reply:
Oh, we understand now. We had it working perfectly, so we didn't understand what the previous reviewer said. Apparently it was working fine in live-server with data provided, but then we saw in another browser all those errors. Should be fixed now, thanks !